stack/tools/src/main/java/org/apache/usergrid/tools/ExportApp.java

```diff
@@ -0,0 +1,552 @@
+        options.addOption( appNameOption );
+
+        Option readThreadsOption = OptionBuilder.hasArg().withType(0)
+                .withDescription( "Read Threads -" + READ_THREAD_COUNT ).create( READ_THREAD_COUNT );
+        options.addOption( readThreadsOption );
+
+        Option writeThreadsOption = OptionBuilder.hasArg().withType(0)
+                .withDescription( "Write Threads -" + WRITE_THREAD_COUNT ).create(WRITE_THREAD_COUNT);
+        options.addOption( writeThreadsOption );
+
+        return options;
+    }
+
+    
+    /**
+     * Tool entry point. 
+     */
+    @Override
+    public void runTool(CommandLine line) throws Exception {
+        
+        applicationName = line.getOptionValue( APPLICATION_NAME );
+
+        if (StringUtils.isNotEmpty( line.getOptionValue( READ_THREAD_COUNT ) )) {
+            try {
+                readThreadCount = Integer.parseInt( line.getOptionValue( READ_THREAD_COUNT ) );
+            } catch (NumberFormatException nfe) {
+                logger.error( "-" + READ_THREAD_COUNT + " must be specified as an integer. Aborting..." );
+                return;
+            }
+        }
+        
+        if (StringUtils.isNotEmpty( line.getOptionValue( WRITE_THREAD_COUNT ) )) {
+            try {
+                writeThreadCount = Integer.parseInt( line.getOptionValue( WRITE_THREAD_COUNT ) );
+            } catch (NumberFormatException nfe) {
+                logger.error( "-" + WRITE_THREAD_COUNT + " must be specified as an integer. Aborting..." );
+                return;
+            }
+        }
+
+        setVerbose( line );
+
+        applyOrgId( line );
+        prepareBaseOutputFileName( line );
+        outputDir = createOutputParentDir();
+        logger.info( "Export directory: " + outputDir.getAbsolutePath() );
+
+        startSpring();
+        
+        UUID applicationId = emf.lookupApplication( applicationName );
+        final EntityManager em = emf.getEntityManager( applicationId );
+        organizationName = em.getApplication().getOrganizationName();
+
+        ExecutorService readThreadPoolExecutor = Executors.newFixedThreadPool( readThreadCount );
+        readScheduler = Schedulers.from( readThreadPoolExecutor );
+
+        ExecutorService writeThreadPoolExecutor = Executors.newFixedThreadPool( writeThreadCount );
+        writeScheduler = Schedulers.from( writeThreadPoolExecutor );
+
+        Observable<String> collectionsObservable = Observable.create( new CollectionsObservable( em ) );
+        
+        collectionsObservable.flatMap( new Func1<String, Observable<ExportEntity>>() {
+            
+            public Observable<ExportEntity> call(String collection) {
+                return Observable.create( new EntityObservable( em, collection ))
+                        .doOnNext( new EntityWriteAction() ).subscribeOn( writeScheduler );
+            }
+            
+        }, 10).flatMap( new Func1<ExportEntity, Observable<ExportConnection>>() {
```

Comment:
Rather than have a static 10 here, we should probably use the Schedulers.io() scheduler, then use the write thread count as this value. Otherwise you'll just introduce contention on a thread pool.  If we use the unbounded Schedulers.io(), and us the size as the flatmap argument, we'll never exceed the set thread size.